blog/2018/12/15/practical-web-content-munging/index.markdown

```diff
@@ -0,0 +1,185 @@
+
+I want to divide those news items out into their own individual pages, which can then be aggregated in whatever way I like, such as having them available in a paginated list or having the most recent items included in a div on the front page of the website.
+
+Those news items look something like this on the old site:
+
+```html
+<h1>Latest News</h1>
+
+<h3>Frobnitz 4.5 released</h3>
+<p>
+This release improves castigation of the widely formed sonterols.
+<br>
+Current users may upgrade by applying coil oil to application points A, C, W, 
+DF, Y0, IN34, RS232, and Frank, then gently inserting the conjubilant apparatus
+into the ferulic treeble socket.
+</p>
+<p class="post-footer align-right">
+<span class="date">December 2, 2018</span>
+</p>
+
+<h3>Frobnitz 4.4 released</h3>
+<p>
+This release is effulgent and wavering gently.
+<br>
+Becomes bees.
+</p>
+<p class="post-footer align-right">
+<span class="date">November 30, 2018</span>
+</p>
+```
+
+Notice that the structure of this is regular but not selectable with any one div or piece of markup. I can use `find('h3')` to get the headings, but the text of each news item is just a paragraph, and we also want to grab the date separately.
+
+So, I want to grab all of the titles, and grab the paragraph following the title, and the date, and put them all into some sort of data structures so I can spit them out into pages of there own.
+
+Let's start with the titles, as it'll show a neat trick Mojo has up its sleeves.
+
+```perl
+my $main = $tx->res->dom->at('#main');
+my @headers = $main->find('h3')->map('text')->each;
+```
+
+Do you see it? The `find` method here is returning a [Mojo::Collection](https://metacpan.org/pod/Mojo::Collection). "Collection" is kind of a fancy way to say "list", but these lists have a bunch of useful utility methods, similar to some core Perl functions that operate on lists, as well as methods found in `List::Util`. It has the usual suspects, like `join`, `grep`, `map`, and `each`. So, collections are fancy, and they deserve a fancy name.
+
+After this `@headers` will now contain all of the titles for all of the news items on the page. Try doing that in two lines of clear, readable, code with regexes (and, we could have chained all of this, including finding `#main`, into one line, but because I'm re-using `#main` multiple times I put it into its own variable).
+
+To break this down a little bit, `map` calls the given method (`text`) on every item returned by `find`, and then `each` returns them all as a list which can be slurped into an array, as I've done.
```

Comment:
This paragraph feels like it should be part of the paragraph with "It has the usual suspects". And overall you can cut down on the basics detail slightly here. You've already linked to brian's article and he does some of it. You're trying to show the practical/real world usage and the more interesting stuff like siblings later. No need to get too into the weeds early on.

Reply:
I think this one is better now, and certainly more concise.